xdsl/transforms/lower_riscv_func.py

```diff
@@ -64,15 +64,15 @@ def match_and_rewrite(self, op: riscv_func.SyscallOp, rewriter: PatternRewriter)
 class LowerRISCVFuncOp(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: riscv_func.FuncOp, rewriter: PatternRewriter):
-        body = op.func_body.block
-        first_op = body.first_op
+        first_block = op.func_body.blocks[0]
+        first_op = first_block.first_op
         assert first_op is not None
```

Comment:
Why asserting `first_op` is not `None`? Could there be a possibility that the first block is empty but subsequent blocks are non-empty for some reason during rewrite?

Reply:
Good question, I don't remember. I think it might be because we didn't have the necessary alternative API for the same logic, today I'd probably write it a little differently